spec/unit/network/formats_spec.rb

```diff
@@ -378,65 +378,126 @@ def self.from_binary(data)
 
-    it "should render empty hashes as empty strings" do
-      expect(console.render({})).to eq('')
+    context "when rendering arrays" do
+      {
+        []                => "",
+        [1, 2]            => "1\n2\n",
+        ["one"]           => "one\n",
+        [{1 => 1}]        => "{1=>1}\n",
+        [[1, 2], [3, 4]]  => "[1, 2]\n[3, 4]\n"
+      }.each_pair do |input, output|
+        it "should render #{input.inspect} as one item per line" do
+          expect(console.render(input)).to eq(output)
+        end
+      end
     end
 
-    it "should render a non-trivially-keyed Hash as pretty printed PSON" do
-      hash = { [1,2] => 3, [2,3] => 5, [3,4] => 7 }
-      expect(console.render(hash)).to eq(PSON.pretty_generate(hash).chomp)
-    end
+    context "when rendering hashes" do
+      {
+        {}                                   => "",
+        {1 => 2}                             => "1  2\n",
+        {"one" => "two"}                     => "one  \"two\"\n", # odd that two is quoted but one isn't
+        {[1,2] => 3, [2,3] => 5, [3,4] => 7} => "{\n  \"[1, 2]\": 3,\n  \"[2, 3]\": 5,\n  \"[3, 4]\": 7\n}",
+        {{1 => 2} => {3 => 4}}               => "{\n  \"{1=>2}\": {\n    \"3\": 4\n  }\n}"
+      }.each_pair do |input, output|
+        it "should render #{input.inspect}" do
+          expect(console.render(input)).to eq(output)
+        end
+      end
 
-    it "should render a {String,Numeric}-keyed Hash into a table" do
-      pson = Puppet::Network::FormatHandler.format(:pson)
-      object = Object.new
-      hash = { "one" => 1, "two" => [], "three" => {}, "four" => object,
-        5 => 5, 6.0 => 6 }
+      it "should render a {String,Numeric}-keyed Hash into a table" do
+        json = Puppet::Network::FormatHandler.format(:json)
+        object = Object.new
+        hash = { "one" => 1, "two" => [], "three" => {}, "four" => object, 5 => 5,
+                 6.0 => 6 }
 
-      # Gotta love ASCII-betical sort order.  Hope your objects are better
-      # structured for display than my test one is. --daniel 2011-04-18
-      expect(console.render(hash)).to eq <<EOT
+        # Gotta love ASCII-betical sort order.  Hope your objects are better
+        # structured for display than my test one is. --daniel 2011-04-18
+        expect(console.render(hash)).to eq <<EOT
 5      5
 6.0    6
-four   #{pson.render(object).chomp}
+four   #{json.render(object).chomp}
 one    1
 three  {}
 two    []
 EOT
-    end
+      end
 
-    it "should render a hash nicely with a multi-line value" do
-      pending "Moving to PSON rather than PP makes this unsupportable."
-      hash = {
-        "number" => { "1" => '1' * 40, "2" => '2' * 40, '3' => '3' * 40 },
-        "text"   => { "a" => 'a' * 40, 'b' => 'b' * 40, 'c' => 'c' * 40 }
-      }
-      expect(console.render(hash)).to eq <<EOT
+      it "should render a hash nicely with a multi-line value" do
+        pending "Moving to PSON rather than PP makes this unsupportable."
```

Comment:
Ah right, I forgot about that. Apparently puppet at one time would use `pp` to emit output like:

```
(byebug) pp datum
{"number"=>
  {"1"=>"1111111111111111111111111111111111111111",
   "2"=>"2222222222222222222222222222222222222222",
   "3"=>"3333333333333333333333333333333333333333"},
 "text"=>
  {"a"=>"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
   "b"=>"bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
   "c"=>"cccccccccccccccccccccccccccccccccccccccc"}}
```

I was thinking of leaving the pending test, and coming back to this issue if I had time. I could also delete the test, since puppet has worked this way since 3.0 at least.